### PR TITLE
SR-IOV topology manager tests

### PR DESCRIPTION
**Pull request by cynepco3hahue**
- provide basic tests that will verify topology manager configuration
- provide SR-IOV test, the kubelet should use the same NUMA node for SR-IOV and CPU's

Tested locally until we will have SR-IOV bare metal environment under CI:
```
Running Suite: Performance Addon Operators e2e tests
====================================================
Random Seed: 1577119246
Will run 2 of 2 specs

• [SLOW TEST:13.194 seconds]
[performance]Topology Manager
/home/alukiano/go/src/github.com/openshift-kni/performance-addon-operators/functests/performance/topology_manager.go:22
  should be enabled with the best-effort policy
  /home/alukiano/go/src/github.com/openshift-kni/performance-addon-operators/functests/performance/topology_manager.go:33
------------------------------
• [SLOW TEST:62.955 seconds]
[performance]Topology Manager
/home/alukiano/go/src/github.com/openshift-kni/performance-addon-operators/functests/performance/topology_manager.go:22
  with the SR-IOV devices and static CPU's
  /home/alukiano/go/src/github.com/openshift-kni/performance-addon-operators/functests/performance/topology_manager.go:46
    should allocate resources from the same NUMA node
    /home/alukiano/go/src/github.com/openshift-kni/performance-addon-operators/functests/performance/topology_manager.go:96
------------------------------

JUnit report was created: /tmp/artifacts/unit_report.xml

Ran 2 of 2 Specs in 132.401 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS
```

### CODE REVIEW

**Comment by fedepaol** [comment]
I'd like to see the `util` package split in different packages where the name declares what the package provides.
It would improve readability on the calling site, like `pods.WaitForDeletion` as opposed to `utils.WaitForPodDeletion`, or `nodes.FilterByResource`.

See https://blog.golang.org/package-names

**Comment by cynepco3hahue** [comment]
> I'd like to see the `util` package split in different packages where the name declares what the package provides.
> It would improve readability on the calling site, like `pods.WaitForDeletion` as opposed to `utils.WaitForPodDeletion`, or `nodes.FilterByResource`.
> 
> See https://blog.golang.org/package-names

Done

File: functests/performance/topology_manager.go [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+		Expect(kubeletConfig.TopologyManagerPolicy).To(Equal(kubeletconfigv1beta1.BestEffortTopologyManagerPolicy))
+	})
+
+	Context("with the SR-IOV devices and static CPU's", func() {
+		var testpod *corev1.Pod
+		var sriovNode *corev1.Node
+
+		BeforeEach(func() {
+			sriovNodes := nodes.FilterByResource(cs, workerRTNodes, testutils.ResourceSRIOV)
+			if len(sriovNodes) < 1 {
```
Comment (fedepaol) [inline functests/performance/topology_manager.go]: Don't we risk skipping if our deployment failed for some reason? Maybe it's better to fail (a bit like what you are doing above with worker-rt nodes). If we want to run the suite against a cluster without this feature, we can drive that through FOCUS / SKIP parameters)

File: functests/performance/topology_manager.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+						"The environment does not have nodes with role %q and available %q resources",
+						testutils.RoleWorkerRT,
+						string(testutils.ResourceSRIOV),
+					),
+				)
+			}
+			sriovNode = &sriovNodes[0]
+
+			var err error
+			if testpod != nil {
```
Comment (fedepaol) [inline functests/performance/topology_manager.go]: How could this happen?

File: functests/test_suite_test.go [hunk 3]
```diff
@@ -24,10 +31,24 @@ func TestTest(t *testing.T) {
...
 		rr = append(rr, reporters.NewJUnitReporter(*junitPath))
 	}
-	RunSpecsWithDefaultAndCustomReporters(t, "Test Suite", rr)
+	RunSpecsWithDefaultAndCustomReporters(t, "Performance Addon Operators e2e tests", rr)
 }
 
 var _ = BeforeSuite(func() {
-	//clients.Setup()
-	// Add here the setup for additional features
+	cs := testutils.NewClientSet("")
```
Comment (fedepaol) [inline functests/test_suite_test.go]: could be clients.New() (with a separate module). In that way only const remains in testutils which can then be switched to testconsts

File: functests/performance/topology_manager.go (hunk 2 above)
Comment (cynepco3hahue) [reply functests/performance/topology_manager.go]: I assumed that we will have more SR-IOV topology manager tests under this context, so in general on the first test under the context `testpod` will be `nil`, but for all following tests it will be not `nil`

File: functests/test_suite_test.go (hunk 3 above)
Comment (cynepco3hahue) [reply functests/test_suite_test.go]: will do

File: functests/performance/topology_manager.go (hunk 2 above)
Comment (fedepaol) [reply functests/performance/topology_manager.go]: Ah you are right :+1:

File: functests/performance/topology_manager.go (hunk 1 above)
Comment (cynepco3hahue) [reply functests/performance/topology_manager.go]: It still unclear how our baremetal CI will look, like will we have a different job for SR-IOV, or will we have a number of different environments under a single job, so we will need to check the existence of SR-IOV devices on tests runtime
Let's leave this check for the near future to ease run of all tests(locally), once we will have CI we can remove it.

Comment (fedepaol) [reply functests/performance/topology_manager.go]: Ok, but please add a big TODO then :-)

Comment (cynepco3hahue) [reply functests/performance/topology_manager.go]: hh ok :)

Comment (cynepco3hahue) [reply functests/performance/topology_manager.go]: done

File: functests/test_suite_test.go (hunk 3 above)
Comment (cynepco3hahue) [reply functests/test_suite_test.go]: done

File: functests/test_suite_test.go [hunk 4]
```diff
@@ -24,10 +32,24 @@ func TestTest(t *testing.T) {
...
 		rr = append(rr, reporters.NewJUnitReporter(*junitPath))
 	}
-	RunSpecsWithDefaultAndCustomReporters(t, "Test Suite", rr)
+	RunSpecsWithDefaultAndCustomReporters(t, "Performance Addon Operators e2e tests", rr)
 }
 
 var _ = BeforeSuite(func() {
-	//clients.Setup()
-	// Add here the setup for additional features
+	cs := testclient.New("")
```
Comment (fedepaol) [inline functests/test_suite_test.go]: Sorry I just noticed, how about making this global and using it in all the tests?
In that case I would have a longer (and more descriptive) name since the scope is wider.

Comment (cynepco3hahue) [reply functests/test_suite_test.go]: what name do you prefer?

Comment (fedepaol) [reply functests/test_suite_test.go]: clientSet maybe?

Comment (cynepco3hahue) [reply functests/test_suite_test.go]: hm maybe instead of global variable we can provide getter for the clientset
```
var clientSet *ClientSet

func Get() *ClientSet {
    if clientSet == nil {
        clientSet = New()
    }
    return clientSet
}
```
WDT?

Comment (fedepaol) [reply functests/test_suite_test.go]: We risk races here I think? Don't remember if ginkgo tests are executed in parallel.
On the other hand, I wouldn't waste a mutex here.

Another option could be calling New() in the package's init() func and having the ClientSet variable exposed. 

something along the line of 
```
package client

var ClientSet ClientSet

func init() {
    // init here
}
```

On the calling site it would just be `client.ClientSet`.

Comment (cynepco3hahue) [reply functests/test_suite_test.go]: done
